llvm/lib/CodeGen/TargetRegisterInfo.cpp

```diff
@@ -222,6 +222,25 @@ TargetRegisterInfo::getMinimalPhysRegClass(MCRegister reg, MVT VT) const {
   return BestRC;
 }
 
+const TargetRegisterClass *TargetRegisterInfo::getCommonMinimalPhysRegClass(
```

Comment:
The two functions are almost identical, can we maybe use a template to contain all the inner logic? The only difference I see that's hard to factor out is `Ty.isValid()` vs `VT == MVT::Other`. Since they're both `bool`s, maybe that could be passed in as a param, and the templating could take care of the rest?

Its very little code, but these kind of need to be kept in sync otherwise.

Reply:
Is it possible to represent default argument value in a template function?

Reply:
I was thinking more that you'd just put all the logic in a new function that was templated, and let each of these just call that internally. Then you can keep your API as is (assuming it needs to be that way for a reason) and the two won't get out of sync. The only thing the original functions need to do is evaluate the boolean, and then return the call to the common impl. 

so Somthing like this, where the current functions are just think wrappers.

```
const TargetRegisterClass *TargetRegisterInfo::getCommonMinimalPhysRegClass(MCRegister Reg1, MCRegister Reg2, MVT VT) const {
  bool isValid = VT == MVT::Other;
  return commonImpl( Reg1, Reg2, VT, isValid);
}
```

Reply:
Done. I used type traits to decide the value, please have a look.